### DISCUSSION TITLE
'server only' or import 'server only'

### DISCUSSION THREAD

**Discussion by RobertHH-IS** [vercel/next.js#72898] [App Router]
I have often seen code and examples where there is just 'server only' at the top.  However the docs imply that you need to npm install the server only package and import 'server only'

Which is correct or are they the same?

**Comment by Mr-Meshky** [comment]
"server only" may be used as an explanation in the code, but `server-only' is a package that must be installed and imported so that the code runs only on the server side
So these two are not the same

**Comment by icyJoseph** [comment]
```js
import 'server-only'`;
```

That package does some funny things, as it defines a `react-server` entry point, which is an empty file, but otherwise, the regular file you'd use, or a client component, throws an error at top level. 


```json
  "exports": {
    ".": {
      "react-server": "./empty.js",
      "default": "./index.js"
    }
  }
```
